src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import React, { useState } from 'react';
+
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getMaxId, getUserById, getValidTitle } from './utils/utils';
+
+const initialTodos: Todo[] = todosFromServer.map((todo) => {
+  const user = usersFromServer.find(({ id }) => id === todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title && !userId) {
+      setHasUserIdError(true);
+      setHasTitleError(true);
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+      return;
+    }
+
+    if (!title) {
+      setHasTitleError(true);
+
+      return;
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+
+      return;
+    }
```

Comment:
Try optimizing this

Imagine how it will look if you have 3 or more possible errors.